airline-backcompat-javaxinject/pom.xml

```diff
@@ -0,0 +1,49 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>airline-parent</artifactId>
+        <groupId>com.github.rvesse</groupId>
+        <version>2.9.0-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>airline-backcompat-javaxinject</artifactId>
```

Comment:
Do you really need that one? People can use `javax.inject:javax.inject:1` if they need the old annotations around.

Reply:
@rvesse have you seen this comment? I see `airline-backcompat-javaxinject` is still around in the latest version.

Reply:
Yep, so while I get your point I like having an explicit dependency in the dependency tree because it makes it clear that we're intentionally pulling in something for backwards compatibility that you might not need.  However I will probably do away with shading and just depend on the old dependency explicitly to make the dependency tree clearer.

The plan will then be to make this `optional` for 2.10.x and ultimately remove it entirely for 3.x

Reply:
But you aware that you will pollute the classpath of people? Meaning they will have `@Inject` in various forms coming? That's definitely not ideal given they will at least sometimes use the wrong one, that will get ignored by their CDI layer.

I think we need to be extremely cautious about that.

Now, if excluding the dependencies from my tree is going to work, I will be able to make it work on my side but I'm a bit worried for people less aware of that.

Reply:
In any case, let me know when you're mostly done with this PR and I'll give it a try in my project.

Reply:
Yeah I get that we're polluting the class path but we've already been doing that with past releases, sometimes in quite subtle ways.  I came back to this issue in earnest because I ran into a use case where trying to wrap an Airline CLI entry point around an embedded JAX-RS 3 application server broke the JAX-RS application because Airline's outdated `jarkarta.inject` annotation was nearer in the dependency tree than the one the JAX-RS runtime we were using was expecting, and broke the server only at runtime due to the way the JAX-RS runtime initialised itself only when the first request arrived.

Excluding the dependencies does work for me in my testing.  I'm going to mark this Ready for Review and have pushed out a 2.9.0-SNAPSHOT build to the Sonatype OSS repository